arangod/IResearch/IResearchView.cpp

```diff
@@ -1532,23 +1527,75 @@ int IResearchView::insert(
     return nullptr;
   }
 
+  auto links = properties.hasKey(StaticStrings::LinksField)
+             ? properties.get(StaticStrings::LinksField)
+             : arangodb::velocypack::Slice::emptyObjectSlice();
+
+  // check link auth as per https://github.com/arangodb/backlog/issues/459
+  if (arangodb::ExecContext::CURRENT) {
+    // check new links
+    for (arangodb::velocypack::ObjectIterator itr(links); itr.valid(); ++itr) {
+      if (!itr.key().isString()) {
+        continue; // not a resolvable collection (invalid jSON)
+      }
+
+      auto collection = vocbase.lookupCollection(itr.key().copyString());
+
+      if (collection
+          && !arangodb::ExecContext::CURRENT->canUseCollection(vocbase.name(), collection->name(), arangodb::auth::Level::RO)) {
+        TRI_set_errno(TRI_ERROR_FORBIDDEN);
+        LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
```

Comment:
AFAIK setting errno is the standard ArangoDB pattern for functions that do not return arangodb::Result otherwise the user will never know the reason for the failure without looking as system logs on the server

Reply:
sure but I thought we try to avoid this for new code